### DISCUSSION TITLE
How to build NSP for debugging?

### DISCUSSION THREAD

**Discussion by TonyGravagno** [taozhi8833998/node-sql-parser#2145] [Q&A]
I'm working on a project with node-sql-parser as a dependency. New errors are occurring with the latest NSP build that did not occur in other projects with the same data. The errors come from mysql.pegjs. I build NSP from source using the package.json 'compile' script (`babel src -d lib`). But that doesn't help with the pegjs source.

When running in debug mode, and stepping through NSP, the pegjs code is compiled to JavaScript, so stepping through that doesn't help to identify the problem.

So, can anyone point to documentation that describes how we can do a full debug from a NSP dependent? Or perhaps document it here and maybe we can add it to the ReadMe or somewhere else?

Do we need to webpack for development after the babel compile?

One of the errors is:

> database     : PARSER_NAME || 'mysql',
> ReferenceError: PARSER_NAME is not defined

Another is a typical "expecting" type error, but as I said, the SQL hasn't changed so something new is happening, and I need to find out exactly where it finds the character that it doesn't like in the query.

Thanks!

**Comment by taozhi8833998** [comment]
`PARSER_NAME` is a global variable, Webpack injects when we build the NSP project.
